lib/instrumentation/trace-context.js

```diff
@@ -34,10 +34,13 @@ function defineLazyProp (obj, prop, fn) {
     enumerable: true,
     get () {
       const value = fn()
-      Object.defineProperty(obj, prop, {
-        enumerable: true,
-        value
-      })
+      if (typeof value !== 'undefined') {
```

Comment:
We don't use `undefined` overwrite protection anywhere else. I don't think this is something to worry much about anymore. For now let's just align with the rest of the application, unless you have a specific concern 😃 

```suggestion
      if (value !== undefined) {
```

Reply:
No reason. That's just how I've always written undefined checks. 🤷‍♂️